recommendation_engine/utils/fileutils.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# This file contains file utilities for loading and writing files to memory.
+# For the cloud upload part see data_store module.
+# Copyright © 2018 Avishkar Gupta
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from recommendation_engine.data_store import data_store_wrapper
+
+
+def load_rating(path=''):
+    """Load the rating matrix as return it as a list-of-lists.
+
+    :path: The local pathname for the rating matrix.
+    :returns: The rating matrix in a list-of-lists format where the
+              list at index i represents the itemeset of the ith user.
+    """
+    rating_file_contents = data_store_wrapper.read_generic_file(path).strip()
+    rating_matrix = []
+    for line in rating_file_contents.split('\n'):
+        this_user_ratings = line.strip().split()
+        if this_user_ratings[0] == 0:
+            this_user_item_list = []
+        else:
+            this_user_item_list = set([int(x) for x in this_user_ratings[1:]])
+        rating_matrix.append(this_user_item_list)
```

Comment:
You're right, the reason this went unnoticed is because in the calling function  we do a simple `len()` check first to prune empty combinations, so all the lists get removed.